src/Streamly/Internal/Unicode/Stream.hs

```diff
@@ -606,38 +637,67 @@ encodeLatin1 = S.map convert
                       "input char codepoint " ++ show codepoint
            else fromIntegral codepoint
 
--- | Like 'encodeLatin1' but silently truncates and maps input characters beyond
+-- | Like 'encodeLatin1'' but silently truncates and maps input characters beyond
 -- 255 to (incorrect) chars in 0-255 range. No error or exception is thrown
 -- when such truncation occurs.
 --
--- /Since: 0.7.0/
+-- /Since: 0.7.0 ("Streamly.Data.Unicode.Stream")/
+--
+-- @since 0.8.0
+{-# INLINE encodeLatin1 #-}
+encodeLatin1 :: (IsStream t, Monad m) => t m Char -> t m Word8
+encodeLatin1 = S.map (fromIntegral . ord)
+
+-- | Like 'encodeLatin1' but drops the input characters beyond 255.
+--
+-- @since 0.8.0
+{-# INLINE encodeLatin1_ #-}
+encodeLatin1_ :: (IsStream t, Monad m) => t m Char -> t m Word8
+encodeLatin1_ = S.map (fromIntegral . ord) . S.filter (<= chr (255))
+
+{-# DEPRECATED encodeLatin1Lax "Please use 'encodeLatin1' instead" #-}
```

Comment:
No need to deprecate this as it is exposed only through a deprecated module.

Reply:
It is not obvious what happened to this function or what to use instead.